### ISSUE TITLE
Uncaught TypeError: Cannot call method 'replace' of undefined

### ISSUE THREAD

**Issue by Pumpapa** [AshesOfOwls/jquery.shapeshift#30]
When I load, I get.
Uncaught TypeError: Cannot call method 'replace' of undefined jquery.min.js:1
st.extend.camelCase jquery.min.js:1
st.extend.css jquery.min.js:3
(anonymous function) jquery.min.js:3
st.extend.access jquery.min.js:1
st.fn.extend.css jquery.min.js:3
Plugin.arrange jquery.shapeshift.js:245
Plugin.render jquery.shapeshift.js:188
Plugin.init jquery.shapeshift.js:78
Plugin jquery.shapeshift.js:52
(anonymous function) jquery.shapeshift.js:629
st.extend.each jquery.min.js:1
st.fn.st.each jquery.min.js:1
$.fn.(anonymous function) jquery.shapeshift.js:621

Then, when I try to drag it either fails altogether or success but generates many:

Uncaught TypeError: Cannot call method 'replace' of undefined jquery.min.js:1
st.extend.camelCase jquery.min.js:1
st.extend.css jquery.min.js:3
(anonymous function) jquery.min.js:3
st.extend.access jquery.min.js:1
st.fn.extend.css jquery.min.js:3
Plugin.arrange jquery.shapeshift.js:245
Plugin.setTargetPosition jquery.shapeshift.js:539
(anonymous function) jquery.shapeshift.js:97
st.event.dispatch jquery.min.js:2
y.handle jquery.min.js:2
st.event.trigger jquery.min.js:2
(anonymous function) jquery.min.js:2
st.extend.each jquery.min.js:1
st.fn.st.each jquery.min.js:1
st.fn.extend.trigger jquery.min.js:2
$container.children.filter.draggable.drag jquery.shapeshift.js:420
e.Widget._trigger jquery-ui.min.js:5
e.widget._trigger jquery-ui.min.js:5
(anonymous function) jquery-ui.min.js:5
e.widget._mouseDrag jquery-ui.min.js:5
(anonymous function) jquery-ui.min.js:5
e.widget._mouseMove jquery-ui.min.js:5
(anonymous function) jquery-ui.min.js:5
_mouseMoveDelegate jquery-ui.min.js:5
st.event.dispatch jquery.min.js:2
y.handle jquery.min.js:2

**Comment by AshesOfOwls** [comment]
Interesting, thats a pretty hefty error message most of which looks unfamiliar to me.

Would it be possible to see the code you wrote?

**Comment by Pumpapa** [comment]
Aaah. This is an issue in my code which just happened to show itself deep in shapeshift/jquery (references to a non-existing element id). Sorry.

**Comment by Pumpapa** [comment]
I keep encountering this issue and I've spent hours whittling it down to this. I believe most of this contributes: if I remove or reorder one child in the container or remove the event listener or one child having differing height all is well.

```
<!doctype html>
<html lang="en">
    <head>
        <style>
            #x { height: 100px; }

            .container {position relative; width:100%; }
            .container div { position:absolute; height:50px; width:80px; background-color:#966; }
            .container div[data-ss-colspan="1"] { width: 80px; }
            .container div[data-ss-colspan="2"] { width: 170px; }
            .container div[data-ss-colspan="3"] { width: 260px; }
            .container div[data-ss-colspan="4"] { width: 350px; }
            .container div[data-ss-colspan="5"] { width: 440px; }
            .container div[data-ss-colspan="6"] { width: 530px; }
            .container div[data-ss-colspan="7"] { width: 620px; }
            .container div[data-ss-colspan="8"] { width: 710px; }
            .container div[data-ss-colspan="9"] { width: 800px; }
            .container div[data-ss-colspan="10"] { width: 890px; }
            .container div[data-ss-colspan="11"] { width: 980px; }
            .container div[data-ss-colspan="12"] { width: 1070px; }
        </style>
        <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.9.0/jquery.min.js"></script>
        <script src="http://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
        <script type="text/JavaScript" src="jquery.shapeshift/core/vendor/jquery.ui.touch-punch.min.js"></script>
        <script type="text/JavaScript" src="jquery.shapeshift/core/jquery.shapeshift.js"></script>
    </head>

    <body>
        <div class="container">
            <div id="x" data-ss-colspan="1">x</div>
            <div id="a"  data-ss-colspan="7">a</div>
            <div id="b"  data-ss-colspan="5">b</div>
            <div id="c"  data-ss-colspan="12">c</div>
        </div>

        <script>
            (function($){

                $('.container').shapeshift({
                    enableDrag: true,
                    enableResize: true,
                    align: "left",
                    minColumns: 12,
                //    gutterY: 10
                });

                function handleFileDrop(e) {
                }

                // Setup the dnd listeners.
                var x = document.getElementById('x');
                x.addEventListener('drop', handleFileDrop, false);

            })(jQuery);
        </script>
    </body>
</html>
```

**Comment by AshesOfOwls** [comment]
kk Will look into tonight, I'm at Railsconf all week so I might be slow getting to this.

**Comment by AshesOfOwls** [comment]
I can't seem to get the same error that you have gotten even after copying and pasting your exact code.

I did however modify it a bit to reflect the way I designed shapeshift to handle drop events in case that helps.

``` html
<!doctype html>
<html lang="en">
    <head>
        <style>
            #x { height: 100px; }

            .container {position relative; width:100%; }
            .container div { position:absolute; height:50px; width:80px; background-color:#966; }
            .container div[data-ss-colspan="1"] { width: 80px; }
            .container div[data-ss-colspan="2"] { width: 170px; }
            .container div[data-ss-colspan="3"] { width: 260px; }
            .container div[data-ss-colspan="4"] { width: 350px; }
            .container div[data-ss-colspan="5"] { width: 440px; }
            .container div[data-ss-colspan="6"] { width: 530px; }
            .container div[data-ss-colspan="7"] { width: 620px; }
            .container div[data-ss-colspan="8"] { width: 710px; }
            .container div[data-ss-colspan="9"] { width: 800px; }
            .container div[data-ss-colspan="10"] { width: 890px; }
            .container div[data-ss-colspan="11"] { width: 980px; }
            .container div[data-ss-colspan="12"] { width: 1070px; }
        </style>
        <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.9.0/jquery.min.js"></script>
        <script src="http://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
        <script type="text/JavaScript" src="jquery.shapeshift/core/vendor/jquery.ui.touch-punch.min.js"></script>
        <script type="text/JavaScript" src="jquery.shapeshift/core/jquery.shapeshift.js"></script>
    </head>

    <body>
        <div class="container">
            <div id="x" data-ss-colspan="1">x</div>
            <div id="a"  data-ss-colspan="7">a</div>
            <div id="b"  data-ss-colspan="5">b</div>
            <div id="c"  data-ss-colspan="12">c</div>
        </div>

        <script>
            (function($){

                $('.container').shapeshift({
                    align: "left",
                    minColumns: 12,
                //    gutterY: 10
                });

                function handleFileDrop(e) {
                }

                // Setup the dnd listeners.
                $('.container').on("ss-rearranged", function(e, selected) {
                  if($(selected).attr("id") === "x") {
                    console.log("Dropped the #x element")
                  }
                })

            })(jQuery);
        </script>
    </body>
</html>
```

Could you let me know if after copying that the error still shows.

**Comment by Pumpapa** [comment]
Interesting

> I can't seem to get the same error that you have gotten even after copying and pasting your exact code.

The snippet is fine in FF & Safari on MacOS, but fails in Chrome, under MacOS, Windows and Ubuntu, the latter even with a fresh install of Chromium (in my current project I need Chrome for file/directory access).

**Comment by AshesOfOwls** [comment]
Crazy since I do my main development under Chrome. I'll take a second look at this today.

**Comment by AshesOfOwls** [comment]
I figured it out!

It appears that this error will occur if you have an element that is larger than the initial container width. You had a twelve column spanning element, and the container could not hold it.

So the solution is to not use a column that is larger than the width of the container.

**Comment by AshesOfOwls** [comment]
Will add a fix for this in 3.0

**Comment by bufke** [comment]
I found a case where this error comes up with a element that is not larger than the container. 
http://jsfiddle.net/8RJ7B/1/

Seems related to the tall height of that one element.

**Comment by AshesOfOwls** [comment]
Interesting, thanks for the heads of Bufke. I have just implemented a possible workaround to this in the 3.0 branch and I will see if the same options you provided break it.

**Comment by AshesOfOwls** [comment]
I think I noticed what was wrong - sorry for the delay in finding this out. I think you accidentally closed one of the elements too early. This fork should work: http://jsfiddle.net/34huZ/

This is all fixed in the 3.0 branch so I will be closing this thread.
